tools/ci_build/github/apple/objectivec/assemble_objc_pod_package.py

```diff
@@ -153,7 +153,7 @@ def path_patterns_as_variable_value(patterns: list[str]):
         "C_POD_NAME": c_pod_config["name"],
         "DESCRIPTION": pod_config["description"],
         "INCLUDE_DIR_LIST": path_patterns_as_variable_value(include_dirs),
-        "IOS_DEPLOYMENT_TARGET": framework_info["IOS_DEPLOYMENT_TARGET"],
+        "IOS_DEPLOYMENT_TARGET": framework_info["iphoneos"]["APPLE_DEPLOYMENT_TARGET"],
```

Comment:
should the objc pod also support MacOS? that could address #18458

Reply:
currently the changes only apply to onnxruntime-c pod I put a TODO as the next step to test and add it to the objc pod in the pr comments.

originally the requirement was from spm and that only loads onnxruntime-c as the binary target.